### PR TITLE
AIP-78 Add docs for backfill

### PR DESCRIPTION
**Pull request by dstandish**
Add docs for backfill

### CODE REVIEW

File: docs/apache-airflow/core-concepts/backfill.rst [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+Backfill
+========
+
+Backfill is when you create runs for past dates of a dag.  Airflow provides a mechanism
+to do this through the CLI and REST API.  You provide a dag, a start date, and an end date,
```
Comment (potiuk) [inline docs/apache-airflow/core-concepts/backfill.rst:22]: Should we leave a TODO for UI here? I guess we will also eventually be able to run backfill via UI?

File: docs/apache-airflow/core-concepts/dag-run.rst [hunk 2]
```diff
@@ -144,21 +144,22 @@ if your DAG performs catchup internally.
...
 This process is known as Backfill.
 
-You may want to backfill the data even in the cases when catchup is disabled. This can be done through CLI.
-Run the below command
+This can be done through API or CLI. For CLI usage, run the command below:
 
 .. code-block:: bash
 
-    airflow dags backfill \
+    airflow backfill create --dag DAG_ID \
```
Comment (potiuk) [inline docs/apache-airflow/core-concepts/dag-run.rst:156]:
```suggestion
    airflow backfill create --dag-id DAG_ID \
```

File: docs/apache-airflow/core-concepts/backfill.rst [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+backfill will consider creating runs for.  Whether or not they will be created
+depends on your chosen reprocessing behavior and the states of any existing
+runs in the range at the time you actually run the backfill.
+
+Example:
+--------
+
+.. code-block:: bash
+
+    airflow backfill create --dag tutorial \
```
Comment (potiuk) [inline docs/apache-airflow/core-concepts/backfill.rst:63]:
```suggestion
    airflow backfill create --dag-id tutorial \
```

File: docs/apache-airflow/start.rst [hunk 4]
```diff
@@ -110,7 +110,7 @@ run the commands below.
     # run your first task instance
     airflow tasks test example_bash_operator runme_0 2015-01-01
     # run a backfill over 2 days
-    airflow dags backfill example_bash_operator \
+    airflow backfill create --dag example_bash_operator \
```
Comment (potiuk) [inline docs/apache-airflow/start.rst:113]:
```suggestion
    airflow backfill create --dag-id example_bash_operator \
```

**Review by potiuk — APPROVED** [review]
Few suggestions. LGTM.

File: docs/apache-airflow/core-concepts/dag-run.rst (hunk 2 above)
Comment (dstandish) [reply docs/apache-airflow/core-concepts/dag-run.rst:156]: that is done in the followup pr https://github.com/apache/airflow/pull/43478

File: docs/apache-airflow/start.rst (hunk 4 above)
Comment (dstandish) [reply docs/apache-airflow/start.rst:113]: done in the followup pr https://github.com/apache/airflow/pull/43478

File: docs/apache-airflow/core-concepts/backfill.rst (hunk 3 above)
Comment (dstandish) [reply docs/apache-airflow/core-concepts/backfill.rst:63]: done in the followup pr https://github.com/apache/airflow/pull/43478

File: docs/apache-airflow/core-concepts/backfill.rst (hunk 1 above)
Comment (dstandish) [reply docs/apache-airflow/core-concepts/backfill.rst:22]: yeah UI portion is planned for Q4. We can update the docs once it's implemented.

**Pull request merged**